editors/sc-ide/widgets/util/docklet.hpp

```diff
@@ -74,7 +74,7 @@ class Docklet : public QObject
         return const_cast<Docklet*>(this)->currentContainer() != mDockWidget;
     }
 
-    void setDetached( bool detached );
+    void setDetached( bool detached, bool visible = true );
```

Comment:
consider renaming this function; a call to "`setDetached( bool, bool )`" is opaque. suggestion - `setDetachedWithVisibility`. i don't think a default value for the second argument is a good idea, it will make calling code harder to understand.